libflux/src/core/semantic/builtins.rs

```diff
@@ -508,6 +508,20 @@ pub fn builtins() -> Builtins<'static> {
                         idx: int
                     ) -> t0
                 "#,
+                "findColumn" => r#"
+                    forall [t0, t1, t2] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        fn: (key: t1) -> bool,
+                        column: string
+                    ) -> [t2]
+                "#,
+                "findRecord" => r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        fn: (key: t1) -> bool,
```

Comment:
I think this should be `fn: (key: t0) -> bool`

Reply:
Yeah I wasn't sure about that ... I based it off tableFind, which does the same thing.

Reply:
Is this the same for both `findColumn` and `findRecord`?

Reply:
I know this is temporary so I won't block, but I did want to add some thoughts. Feel free to take them or leave them.

First, `t0` is supposed to represent the group key, however that is not enforced in the code. There's nothing preventing a user from referencing a column in their lambda that is not part of the group key. We don't have any tests for this case either which is concerning. Most likely an error is thrown from somewhere in the compiler package. But in some cases the query probably executes just fine.

Second, I see `findColumn` as just a `map`. In fact, if you look at the type signature of `map`, `findColumn` should be equivalent to:
```
// return the _value column as an array
|> ... map(fn: (r) => r._value)
```
Unfortunately `map` is specialized to table streams, but it would be great if the query engine treated arrays and table streams as interchangeable. Because this is not yet the case, we would have to implement `findColumn` separately similar to how we do now. However a major benefit of giving `findColumn` the same type signature as `map` is that now there's no more return type polymorphism and any semantic errors made on the part of the user will be caught and identified by the type checker.

Reply:
@jlapacik @nathanielc I'm understanding that we need to say  that `t1` is the group-key subset of `t0`, but we have no facility to express that.

Is it true that if we do make them the same type we are doing better than the current solution because we are restricting to columns in the table, vs allowing `key` to be anything?

To be honest I don't fully understand the issues here, as I don't have a very deep understanding of the relationship between the query engine and the runtime of flux.

I can say though that `tableFind` has been doing this same thing since it's inception and that is actively used by customers. So either it's safe, or we have subtly buggy code out there. In either case it warrants more investigation I think, at least for my understanding.

Reply:
@fchikwekwe yeah they both do the same thing, use one type for the table and another type for the records of the table. They were both based off `tableFind`.  It certainly seems wrong because they should be bound to each other, but I don't think we have a way to fix it properly right now.

Reply:
@nathanielc @jlapacik I'm going to merge this on the grounds that it is identical to `tableFind` but I would like to investigate further.

Reply:
That's fine. Like I said I'm not going block. It does concern me a little bit that we don't enforce the argument to `fn` as being part of the group key and only the group key. That is why I think it would be better to substitute `t1` with `t0`. But you're right, it's identical to tableFind's current behavior and so it is not introducing any regressions.